nixos/modules/security/acme/default.nix

```diff
@@ -325,6 +325,7 @@ let
         '');
       } // optionalAttrs (data.listenHTTP != null && toInt (elemAt (splitString ":" data.listenHTTP) 1) < 1024) {
         CapabilityBoundingSet = [ "CAP_NET_BIND_SERVICE" ];
+        AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" ];
```

Comment:
It's my understanding that `CapabilityBoundingSet` and `AmbientCapabilities` are exact opposites of each other -- that is, `CapabilityBoundingSet` is a list of capabilities that shouldn't ever be granted to the process, while `AmbientCapabilities` is a list of capabilities to grant to the service (see [this SO answer](https://unix.stackexchange.com/a/581337)).

This being said, we should remove the `CapabilityBoundingSet` definition here, as it only serves to confuse.

(Do you know why it was even set in the first place? It's the exact opposite of what we want in this condition, it seems. Guessing it was just a mistake and nobody actually used it until now, or something?)

Reply:
They aren't exact opposites, no. CapabilityBoundingSet controls what caps a program is allowed to utilise even when running as root. It is a whitelist of capabilities, and defaults to allowing all if unset. Ambient capabilities grants caps that are otherwise root reserved to non root executions. Thus both still have their place.

It was likely added as part of the service hardening that was conducted at some point. I don't have a link to the commit or PR handy.

Reply:
Thanks for the explanation, and sorry for the misunderstanding.

Reply:
To be fair the wording in that SO post is somewhat ambiguous.

Reply:
AmbientCapabilities was added because of #191794

Reply:
> AmbientCapabilities was added because of #191794

I'm talking about the other option, which was added in place of AmbientCapabilities in 65f1b8c6ae2f2cf6a13d77b98b42eba31eef0424

@m1cr0man Shouldn't the test in that PR have caught this earlier? 🤔